packer_test/provisioner_tests/hcp-sbom/priovisioner_test.go

```diff
@@ -75,11 +75,65 @@ func (ts *PackerHCPSbomTestSuite) TestDestFile_WithIntermediateDirs() {
 
 	ts.PackerCommand().UsePluginDir(dir).
 		SetArgs("build", "./templates/dest_is_file_no_interm_dirs.pkr.hcl").
-		Assert(check.MustSucceed(), check.FileExists("sbom/sbom_cyclonedx", false))
+		Assert(check.MustSucceed(), check.FileExists("sbom/sbom_cyclonedx.json", false))
 
 	os.RemoveAll("sbom")
 }
 
 // * output directory (without trailing slash) - directory exists
+func (ts *PackerHCPSbomTestSuite) TestDestDir_NoTrailingSlash() {
+	ts.SkipNoAcc()
+
+	dir := ts.MakePluginDir()
+	defer dir.Cleanup()
+
+	os.MkdirAll("sbom", 0755)
+
+	ts.PackerCommand().UsePluginDir(dir).
+		SetArgs("plugins", "install", "github.com/hashicorp/docker").
+		Assert(check.MustSucceed())
+
+	ts.PackerCommand().UsePluginDir(dir).
+		SetArgs("build", "./templates/dest_is_dir.pkr.hcl").
+		Assert(check.MustSucceed(), check.FileInDir("sbom", "packer-user-sbom-.*.json"))
```

Comment:
I see, the only difference here is the glob support? I would suggest renaming the checker to `FileGlob`, and unifying the path so the dir and the file glob don't need to be two different arguments, the glob can then be changed into something like `sbom/packer-user-sbom*json`